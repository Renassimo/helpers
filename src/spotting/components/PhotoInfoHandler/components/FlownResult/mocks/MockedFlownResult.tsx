const MockedFlownResult = jest.fn(() => <div>MockedFlownResult</div>);

export default MockedFlownResult;
