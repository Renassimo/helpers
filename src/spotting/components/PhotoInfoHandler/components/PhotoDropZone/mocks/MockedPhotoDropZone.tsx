const MockedPhotoDropZone = jest.fn(() => <div>MockedPhotoDropZone</div>);

export default MockedPhotoDropZone;
