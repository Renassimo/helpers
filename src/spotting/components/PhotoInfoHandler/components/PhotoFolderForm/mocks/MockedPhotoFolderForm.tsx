const MockedPhotoFolderForm = jest.fn(() => <div> MockedPhotoFolderForm</div>);

export default MockedPhotoFolderForm;
