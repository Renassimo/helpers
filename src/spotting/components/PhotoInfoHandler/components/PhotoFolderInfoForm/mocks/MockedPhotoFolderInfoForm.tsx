const MockedPhotoFolderInfoForm = jest.fn(() => (
  <div>MockedPhotoFolderInfoForm</div>
));

export default MockedPhotoFolderInfoForm;
