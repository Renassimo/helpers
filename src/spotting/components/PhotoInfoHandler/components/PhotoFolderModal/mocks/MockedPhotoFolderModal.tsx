const MockedPhotoFolderModal = jest.fn(() => <div>MockedPhotoFolderModal</div>);

export default MockedPhotoFolderModal;
