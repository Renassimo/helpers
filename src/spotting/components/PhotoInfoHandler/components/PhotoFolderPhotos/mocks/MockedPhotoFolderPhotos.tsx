const MockedPhotoFolderPhotos = jest.fn(() => (
  <div>MockedPhotoFolderPhotos</div>
));

export default MockedPhotoFolderPhotos;
