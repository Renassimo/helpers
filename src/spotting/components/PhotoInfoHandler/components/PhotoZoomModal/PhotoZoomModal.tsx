import { useCallback } from 'react';

import Box from '@mui/material/Box';

import { PhotoActionType } from '@/spotting/types';

import SimpleModal from '@/common/components/Modal/SimpleModal';

import usePhotoInfoContext from '@/spotting/contexts/hooks/usePhotoInfoContext';

const PhotoZoomModal = () => {
  const { zoomedPhoto, dispatch } = usePhotoInfoContext();

  const onClose = useCallback(
    () => dispatch({ type: PhotoActionType.CLEAR_ZOOMED_PHOTO }),
    []
  );

  return (
    <SimpleModal
      open={!!zoomedPhoto}
      onClose={onClose}
      title={zoomedPhoto?.path || ''}
      maxWidth="xl"
    >
      <Box>
        <img
          src={zoomedPhoto?.preview}
          alt={zoomedPhoto?.path}
          style={{ width: '100%' }}
        />
      </Box>
    </SimpleModal>
  );
};

export default PhotoZoomModal;
