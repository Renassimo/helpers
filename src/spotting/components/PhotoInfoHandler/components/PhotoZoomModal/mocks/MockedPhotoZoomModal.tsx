const MockedPhotoZoomModal = jest.fn(() => <div>MockedPhotoZoomModal</div>);

export default MockedPhotoZoomModal;
