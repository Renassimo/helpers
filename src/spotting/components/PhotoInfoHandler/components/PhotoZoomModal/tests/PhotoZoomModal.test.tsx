import renderWithTheme from '@/common/tests/helpers/renderWithTheme';

import SimpleModal from '@/common/components/Modal/SimpleModal';

import usePhotoInfoContext from '@/spotting/contexts/hooks/usePhotoInfoContext';

import MockedModal from '@/common/components/Modal/mocks';
import { mockedPhoto } from '@/spotting/types/mocks';

import PhotoZoomModal from '../PhotoZoomModal';

jest.mock('@/common/components/Modal/SimpleModal');
jest.mock('@/spotting/contexts/hooks/usePhotoInfoContext');

describe('PhotoZoomModal', () => {
  const dispatch = jest.fn();

  beforeEach(() => {
    (SimpleModal as unknown as jest.Mock).mockImplementation(MockedModal);
    (usePhotoInfoContext as unknown as jest.Mock).mockImplementation(
      jest.fn(() => ({ dispatch, zoomedPhoto: mockedPhoto }))
    );
  });

  test('renders successfully', () => {
    // Arange
    // Act
    const { baseElement } = renderWithTheme(<PhotoZoomModal />);
    // Assert
    expect(baseElement).toMatchSnapshot();
  });
});
