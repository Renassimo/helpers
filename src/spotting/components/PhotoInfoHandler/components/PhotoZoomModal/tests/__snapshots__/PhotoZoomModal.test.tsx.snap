// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoZoomModal renders successfully 1`] = `
<body>
  <div>
    <div>
      <div
        class="MuiBox-root css-0"
      >
        <img
          alt="path1"
          src="preview1"
          style="width: 100%;"
        />
      </div>
      <div>
        title: 
        path1
        ; maxWidth: 
        xl
        ;
        open
        ;
        not loading
        ;
        not disabled
        ; onClose
         
        passed
        ; onSubmit
         
        did not passed
        ;
      </div>
    </div>
  </div>
</body>
`;
